src/library/coverartutils.h

```diff
@@ -36,9 +36,7 @@ class CoverArtUtils {
     // trackLocation.
     static QImage extractEmbeddedCover(const QString& trackLocation,
                                        SecurityTokenPointer pToken) {
-        if (trackLocation.isEmpty()) {
-            return QImage();
-        }
+        DEBUG_ASSERT(!trackLocation.isEmpty());
```

Comment:
Empty file names and non-existent files are handled gracefully by SoundSourceProxy, we don't need to handle these errors here again. Instead of handling it twice I would prefer to remove the DEBUG_ASSERT entirely.